indigo_app/views.py

```diff
@@ -261,6 +269,142 @@ def post(self, request, frbr_uri, version_id):
         return redirect(url)
 
 
+class BatchAddWorkView(AbstractAuthedIndigoView, TemplateView):
+    template_name = 'work/new_batch.html'
+    # permissions
+    permission_required = ('indigo_api.add_work',)
+
+    def get_context_data(self, **kwargs):
+        context = super(BatchAddWorkView, self).get_context_data(**kwargs)
+
+        def get_works():
+
+            table = get_table()
+            works = []
+
+            for idx, row in enumerate(table):
+                # TODO: match header row contents rather than using column numbers
+                row_number = idx+1
+
+                info = {
+                    'row': row_number,
+                }
+
+                frbr_uri = get_frbr_uri(row)
+
+                # TODO: fix get_uri to return uri as false if validation error?
+
+                if frbr_uri:
+
+                    try:
+                        work = Work.objects.get(frbr_uri=frbr_uri)
+                        info['work'] = work
+                        info['status'] = 2
```

Comment:
Magic numbers like this can be confusing because it's not immediately apparent what they mean. In some languages, one could define a constant like `EXISTS = 2` and then use `EXISTS` rather than 2, which makes it more legible. In python (and in this case), you could also use a string, such as `info['status'] = 'exists'`.